File: ems_frontend/src/api/device.ts
import axios from "../helper/axios";

// Types
export interface Device {
  id: number;
  sn: string;
  create_id: number;
  create_time: string;
  modify_id: number;
  modify_time: string;
}

export interface DeviceCreateRequest {
  sn: string;
}

export interface DeviceUpdateRequest {
  sn: string;
}

// API functions
export const fetchDevices = async () => {
  try {
    const response = await axios.get("/devices");
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const fetchDeviceById = async (id: number) => {
  try {
    const response = await axios.get(`/devices/${id}`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const createDevice = async (payload: DeviceCreateRequest) => {
  try {
    const response = await axios.post("/devices", payload);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const updateDevice = async (id: number, payload: DeviceUpdateRequest) => {
  try {
    const response = await axios.put(`/devices/${id}`, payload);
    return response.data;
  } catch (error) {
    throw error;
  }
};

export const deleteDevice = async (id: number) => {
  try {
    const response = await axios.delete(`/devices/${id}`);
    return response.data;
  } catch (error) {
    throw error;
  }
};

// 獲取未被綁定到任何公司的設備
export const fetchUnassignedDevices = async () => {
  try {
    const response = await axios.get("/devices/unassigned");
    return response.data;
  } catch (error) {
    throw error;
  }
};

File: ems_backend/internal/domain/device/entities/device.go
package entities

import "time"

// Device 設備實體
type Device struct {
	ID         uint
	SN         string
	CreateID   uint
	CreateTime time.Time
	ModifyID   uint
	ModifyTime time.Time
}

// NewDevice 創建新設備
func NewDevice(sn string, createID uint) *Device {
	now := time.Now()
	return &Device{
		SN:         sn,
		CreateID:   createID,
		CreateTime: now,
		ModifyID:   createID,
		ModifyTime: now,
	}
}

// Update 更新設備
func (d *Device) Update(sn string, modifyID uint) {
	d.SN = sn
	d.ModifyID = modifyID
	d.ModifyTime = time.Now()
}
